pkg/apis/machine/v1beta1/machine_webhook_test.go

```diff
@@ -956,6 +956,65 @@ func TestValidateAWSProviderSpec(t *testing.T) {
 			expectedOk:       true,
 			expectedWarnings: []string{"providerSpec.iamInstanceProfile: no IAM instance profile provided: nodes may be unable to join the cluster"},
 		},
+		{
+			testCase: "with double tag names, lists duplicated tags",
+			modifySpec: func(p *aws.AWSMachineProviderConfig) {
+				p.Tags = []aws.TagSpecification{
+					{
+						Name: "Tag-A",
+					},
+					{
+						Name: "Tag-B",
+					},
+					{
+						Name: "Tag-C",
+					},
+					{
+						Name: "Tag-A",
+					},
+					{
+						Name: "Tag-B",
+					},
+				}
+			},
+			expectedOk:       true,
+			expectedWarnings: []string{"providerSpec.tags: duplicated tag names (Tag-A,Tag-B): only the first value will be used."},
```

Comment:
the only question i have about this given the description in the pr, should we be warning about non-duplicated tags which will be ignored? (unless i am misunderstanding how this works)

Reply:
Do you mean if a customer sets a tag such as `Name` which we later filter out? That's a possibility, we could add that in as well

Reply:
right, i guess i was thinking about this test where it seems like "Tag-C" would also be filtered. not due to duplicate, just because it is extra. assuming that is correct, it seems like we would want to warn on all filtered tags.

Reply:
I don't think Tag C is being filtered out here? I think the point of the list warning here is to find duplicated tags, we omit tag C from the list because it only exists once. The warning should only contain those that are duplicated IMO, not all tags, though that could be changed if you disagree

Reply:
ok, disregard my comment then. i was confused about the logic, i thought that tag-c _was_ getting filtered as well.